applications/GeoMechanicsApplication/custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp

```diff
@@ -0,0 +1,810 @@
+            "convergence_criteria_settings"       : {},
+            "linear_solver_settings"              : {},
+            "scheme_settings"                     : {}
+        })");
+
+        // Getting base class default parameters
+        const Parameters base_default_parameters = BaseType::GetDefaultParameters();
+        default_parameters.RecursivelyAddMissingParameters(base_default_parameters);
+        return default_parameters;
+    }
+
+    /**
+     * @brief Returns the name of the class as used in the settings (snake_case format)
+     * @return The name of the class
+     */
+    static std::string Name() { return "newton_raphson_strategy_linear_elastic_dynamic"; }
+
+    ///@}
+    ///@name Operators
+
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ResidualBasedNewtonRaphsonStrategyLinearElasticDynamic";
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+private:
+    ///@}
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    double mBeta;
+    double mGamma;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    /**
+     * @brief This method prints information after reach the max number of iterations
+     */
+    void MaxIterationsExceeded() override
+    {
+        KRATOS_INFO_IF("ResidualBasedNewtonRaphsonStrategyLinearElasticDynamic", BaseType::GetEchoLevel() > 0)
+            << "ATTENTION: max iterations ( " << BaseType::mMaxIterationNumber << " ) exceeded!"
+            << std::endl;
+    }
+
+    bool PerformIterationCycle(TSystemMatrixType&   rA,
+                               TSystemVectorType&   rDx,
+                               TSystemVectorType&   rb,
+                               TSystemVectorType&   rDxTot,
+                               std::vector<Vector>& rNonconvergedSolutions,
+                               unsigned int&        rIterationNumber)
+    {
+        ModelPart&                    r_model_part = BaseType::GetModelPart();
+        typename TSchemeType::Pointer p_scheme     = BaseType::GetScheme();
+        typename TBuilderAndSolverType::Pointer p_builder_and_solver = BaseType::GetBuilderAndSolver();
+        auto&       r_dof_set              = p_builder_and_solver->GetDofSet();
+        const auto& r_current_process_info = r_model_part.GetProcessInfo();
+
+        bool is_converged = false;
+        rIterationNumber++;
+        for (; rIterationNumber < BaseType::mMaxIterationNumber; rIterationNumber++)
+        {
+            // setting the number of iteration
+            r_model_part.GetProcessInfo()[NL_ITERATION_NUMBER] = rIterationNumber;
+
+            p_scheme->InitializeNonLinIteration(r_model_part, rA, rDx, rb);
```

Comment:
This initialize still does both the elements and conditions, is that intentional?

Reply:
Just checked, here the elements need to be initialized (otherwise we get different results in the end)